catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-converter/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_1_0/catalog/converter/impl/GenericFeatureConverterWfs11.java

```diff
@@ -0,0 +1,134 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.ogc.wfs.v1_1_0.catalog.converter.impl;
+
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Date;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.spatial.ogc.wfs.catalog.common.WfsConstants;
+import org.codice.ddf.spatial.ogc.wfs.catalog.mapper.MetacardMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class works in conjunction with XStream to convert a {@link Metacard} to XML according to
+ * the GML 3.1.1 spec. It will also convert respective XML into a Metacard.
+ */
+public class GenericFeatureConverterWfs11 extends AbstractFeatureConverterWfs11 {
+
+  private static final String ID = "id";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GenericFeatureConverterWfs11.class);
+
+  private String sourceId = null;
+
+  public GenericFeatureConverterWfs11(String srs) {
+    super(srs);
+  }
+
+  public GenericFeatureConverterWfs11(MetacardMapper metacardMapper) {
+    super(metacardMapper);
+  }
+
+  /**
+   * Method to determine if this converter knows how to convert the specified Class.
+   *
+   * @param clazz the class to check
+   */
+  @Override
+  public boolean canConvert(Class clazz) {
+    return Metacard.class.isAssignableFrom(clazz);
+  }
+
+  @Override
+  public void marshal(Object value, HierarchicalStreamWriter writer, MarshallingContext context) {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * This method will unmarshal an XML instance of a "gml:member" to a {@link Metacard}.
+   *
+   * @param hreader the stream reader responsible for reading this xml doc
+   * @param context a reference back to the Xstream unmarshalling context. Allows you to call
+   *     "convertAnother" which will lookup other registered converters.
+   */
+  @Override
+  public Object unmarshal(HierarchicalStreamReader hreader, UnmarshallingContext context) {
+
+    LOGGER.debug("Entering: {} : unmarshal", this.getClass().getName());
+    // Workaround for Xstream which seems to be having issues involving attributes with namespaces,
+    // in that it cannot fetch the attributes value directly by name.
+    String id = null;
+    int count = hreader.getAttributeCount();
+    for (int i = 0; i < count; ++i) {
```

Comment:
Could we use a post increment here to avoid confusion? The effect is the same but I really had to think about it to confirm that.

Reply:
done